web/src/main/java/org/springframework/security/web/bind/support/CurrentSecurityContextArgumentResolver.java

```diff
@@ -0,0 +1,172 @@
+ * {@link ClassCastException} will be thrown.
+ * </p>
+ *
+ * @author Dan Zheng
+ * @since 5.2
+ */
+public final class CurrentSecurityContextArgumentResolver
+		implements HandlerMethodArgumentResolver {
+
+	private ExpressionParser parser = new SpelExpressionParser();
+
+	private BeanResolver beanResolver;
+	/**
+	 * check if this argument resolve can support the parameter.
+	 * @param parameter the method parameter.
+	 * @return true = it can support parameter.
+	 *
+	 * @see
+	 * org.springframework.web.method.support.HandlerMethodArgumentResolver#
+	 * supportsParameter(org.springframework.core.MethodParameter)
+	 */
+	public boolean supportsParameter(MethodParameter parameter) {
+		return findMethodAnnotation(CurrentSecurityContext.class, parameter) != null;
+	}
+
+	/**
+	 * resolve the argument to inject into the controller parameter.
+	 * @param parameter the method parameter.
+	 * @param mavContainer the model and view container.
+	 * @param webRequest the web request.
+	 * @param binderFactory the web data binder factory.
+	 *
+	 * @see org.springframework.web.method.support.HandlerMethodArgumentResolver#
+	 * resolveArgument (org.springframework.core.MethodParameter,
+	 * org.springframework.web.method.support.ModelAndViewContainer,
+	 * org.springframework.web.context.request.NativeWebRequest,
+	 * org.springframework.web.bind.support.WebDataBinderFactory)
+	 */
+	public Object resolveArgument(MethodParameter parameter,
+				ModelAndViewContainer mavContainer, NativeWebRequest webRequest,
+				WebDataBinderFactory binderFactory) throws Exception {
+		SecurityContext securityContext = SecurityContextHolder.getContext();
+		if (securityContext == null) {
+			return null;
+		}
+		Object securityContextResult = securityContext;
+
+		CurrentSecurityContext securityContextAnnotation = findMethodAnnotation(
+				CurrentSecurityContext.class, parameter);
+
+		String expressionToParse = securityContextAnnotation.expression();
+		if (StringUtils.hasLength(expressionToParse)) {
+			StandardEvaluationContext context = new StandardEvaluationContext();
+			context.setRootObject(securityContext);
+			context.setVariable("this", securityContext);
+
+			Expression expression = this.parser.parseExpression(expressionToParse);
+			securityContextResult = expression.getValue(context);
+		}
+
+		if (securityContextResult != null
+				&& !parameter.getParameterType().isAssignableFrom(securityContextResult.getClass())) {
+			if (securityContextAnnotation.errorOnInvalidType()) {
+				throw new ClassCastException(securityContextResult + " is not assignable to "
+						+ parameter.getParameterType());
+			}
+			else {
+				return null;
+			}
+		}
+		return securityContextResult;
+	}
+	/**
+	 * Sets the {@link BeanResolver} to be used on the expressions
+	 * @param beanResolver the {@link BeanResolver} to use
+	 */
+	public void setBeanResolver(BeanResolver beanResolver) {
+		this.beanResolver = beanResolver;
```

Comment:
Please check for `null` here using something like:

```java
Assert.notNull(beanResolver, "beanResolver cannot be null");
```

Reply:
fixed